src/app/pages/categories/management/categories-editor/pages-categories-editor.component.ts

```diff
@@ -0,0 +1,106 @@
+import { Component, Inject, OnInit } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dialog';
+import { Observable, of } from 'rxjs';
+import { checkInputError } from 'src/app/common/utils/forms/common-utils-forms-form-utils';
+import { SystemNotificationsService } from 'src/app/common/utils/system-notifications/system-notifications.service';
+import { NotificationType } from 'src/app/common/utils/system-notifications/system.notifications.constants';
+import { WalletTransactionType } from 'src/app/domains/transactions/domains.transactions.constants';
+import { PagesTransactionCategoriesService } from '../../pages-transaction-categories.service';
+import { TransactionCategoryFull } from '../../transaction-category-full.model';
+import { TransactionCategory } from '../../transaction-category.model';
+import { ITransactionCategoryEditorPayload } from './pages-categories-editor.service';
+import { ITransactionCategoryModalFormType } from './pages-categories-editor.types';
+
+@Component({
+  selector: 'app-pages-categories-editor',
+  templateUrl: './pages-categories-editor.component.html',
+  styleUrls: [ './pages-categories-editor.component.scss' ],
+})
+export class PagesCategoriesEditorComponent implements OnInit {
+  public selectTransactionsTypes: Record<string, WalletTransactionType> = {
+    'Income': WalletTransactionType.Income,
+    'Expense': WalletTransactionType.Expense,
+  };
+
+  public form!: FormGroup<ITransactionCategoryModalFormType>;
```

Comment:
```suggestion
  public form: FormGroup<ITransactionCategoryModalFormType> | null = null;
```

It should be typed in such way that we know that initially form is not there. If it would be created in `ngOnInit` without request then sure, we can use `!` because it would be created as soon as component is created (basically like in constructor - in such case TS is not requiring `!` at all). But in case of asynchronous creation of form we need to know that initially it is not available.